lm_eval/evaluator.py

```diff
@@ -469,18 +472,19 @@ def evaluate(
 
         ### Aggregate results over all datapoints ###
         # aggregate results ; run bootstrap CIs
-        for (task_name, key, metric), items in vals.items():
+        # for (task_name, key, metric), items in vals.items():
+        for (task_name, key), items in vals.items():
             task = task_dict[task_name]
-            metric_key = metric + "," + key
+            # metric_key = metric + "," + key
 
             if type(task) == tuple:
                 group_name, task = task
             else:
                 group_name = None
 
-            agg_fn = task.aggregation()[metric]
-            results[task_name][metric_key] = agg_fn(items)
-            results[task_name]["samples"] = len(items)
+            for metric_key, metric_fn in task.aggregation().items():
+                results[task_name][metric_key] = metric_fn(*list(zip(*items)))
+                results[task_name]["samples"] = len(items)
```

Comment:
@haileyschoelkopf @baberabb 

For the PR, metrics calculation (both the per sample and aggregation) will be moved here. I think for the `--predict_only` we can opt to exit the with the model output before this and that way we won't need to bypass metrics.